airflow/providers/cloudant/provider.yaml

```diff
@@ -46,7 +47,17 @@ versions:
 
 dependencies:
   - apache-airflow>=2.8.0
-  - cloudant>=2.13.0
+  - ibmcloudant>=0.9.1
+
+excluded-python-versions:
+  # ibmcloudant transitively brings in urllib3 2.x, but the snowflake provider has a dependency that pins
+  # urllib3 to 1.x on Python 3.8 and 3.9; thus we exclude those Python versions from taking the update
+  # to ibmcloudant.
+  # See #21004, #41555, and https://github.com/snowflakedb/snowflake-connector-python/issues/2016
+  - "3.8"
+  - "3.9"
+  # ibmcloudant brings in ibm-cloud-sdk-core which does not support Python 3.12
+  - "3.12"
```

Comment:
Can you clarify about this?
I see it does: https://github.com/IBM/python-sdk-core/blob/d603ed0476fb3ceca0b1e82ae1259d3725e2948f/pyproject.toml#L17

Reply:
Yes I can calrify - 3.12 support was added to ibm-cloud-sdk-core in 3.20.5, the latest ibmcloundant pins ibm-cloud-sdk-core at 3.20.3. (I don't think anything actually changed in the code or deps, they just added 3.12 to the classifiers.)

Reply:
Is there an issue or pr we can track progress on this?

Reply:
ibmcloudant has already [bumped ibm-cloud-sdk-core to 3.20.6](https://github.com/IBM/cloudant-python-sdk/commit/ff052f7ce1280d53954dbd07a62f8f580fc7f65a), however they haven't released yet. I can't find an official release schedule in the repo, but their releases look to be timed 4-8 weeks apart so a new one should be coming soon.

Would you like me to file an issue in the Airflow project or is a note/TODO in the provider.yaml sufficient?

Reply:
Todo is needed any way. The issue is that it should have something simple to track (issue/pr) we should avoid leaveing exploration tasks. I think filing an issue is simple enough and will provide the clarity we need

Reply:
I removed the 3.12 exclusion as I think the test failures were from apache beam and other issues, not the ibmcloudant dependencies. If tests pass we know it's gtg on 3.12.